### PR TITLE
entity filtering feature for reactive option

### PR DESCRIPTION
**Pull request by dwarakaprasad**
ConditionBuilder class to build conditions from EntityCriteria

Related with jhipster/generator-jhipster#14616
Related to jhipster/generator-jhipster#19959

### CODE REVIEW

**Comment by gzsombor** [comment]
Great patch, I see just a couple of minor things

**Review by gzsombor — CHANGES_REQUESTED** [review]

File: jhipster-framework/src/main/java/tech/jhipster/service/ConditionBuilder.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+ */
+public class ConditionBuilder {
+
+    private final List<Condition> allFilters = new ArrayList<Condition>();
+
+    private final ColumnConverterReactive columnConverter;
+
+    private ConditionBuilder() {
+        this.columnConverter = null;
+    }
```
[the remark this answers is not in the document]
Comment (dwarakaprasad) [reply jhipster-framework/src/main/java/tech/jhipster/service/ConditionBuilder.java:53]: Done

File: jhipster-framework/src/main/java/tech/jhipster/service/ConditionBuilder.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+                        .getNotIn()
+                        .stream()
+                        .map(eachNotIn -> SQL.literalOf(columnConverter.convert(eachNotIn, Boolean.class)))
+                        .collect(Collectors.toList())
+                )
+            );
+        }
+        if (generalData.getSpecified() != null && generalData.getSpecified()) {
+            // There is no IS or isNotNull method to create this equivalent
+            allFilters.add(Conditions.just("" + column + " IS NOT NULL"));
```
[the remark this answers is not in the document]
Comment (dwarakaprasad) [reply jhipster-framework/src/main/java/tech/jhipster/service/ConditionBuilder.java:186]: Done

File: jhipster-framework/src/test/java/tech/jhipster/service/demo/ConditionBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+		assert (builder.buildConditions().toString()).contentEquals("c.time_to_go != '21600000'");
+	}
+
+	@Test
+	void testUsingDurationTypeWithAllFilters()
+			throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+		DurationFilter testDuration = (DurationFilter) new ConditionTestHelper(rangeFilterSetters, Arrays.asList(),
+				generalFilterSetters, true).buildRangeConditions(new DurationFilter(), TEST_DURATION);
+		setupBuilder(testDuration);
+		assert (builder.buildConditions().toString())
```
[the remark this answers is not in the document]
Comment (dwarakaprasad) [reply jhipster-framework/src/test/java/tech/jhipster/service/demo/ConditionBuilderTest.java:117]: Done

File: jhipster-framework/src/main/java/tech/jhipster/service/ConditionBuilder.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+            .stream()
+            .reduce(
+                null,
+                (Condition cumulated, Condition eachCondition) -> {
+                    return cumulated != null ? cumulated.and(eachCondition) : eachCondition;
+                }
+            );
+    }
+
+    private <X extends Comparable<? super X>> void buildRangeConditions(RangeFilter<X> rangeData, Column column, Class<?> targetClass) {
```
[the remark this answers is not in the document]
Comment (dwarakaprasad) [reply jhipster-framework/src/main/java/tech/jhipster/service/ConditionBuilder.java:105]: Done

**Comment by DanielFran** [comment]
@gzsombor Your requested changes have been impleted. Can you confirm we can merge the PR?
Thanks

**Comment by gzsombor** [comment]
Sorry, I haven't had time, but I still see, that on the values, toString is called which makes me worried. I don't know by heart, how the R2DBC drivers, and the database are handling types, but I would be generally surprised that all - with all possible locale settings ordering/filtering would work correctly, when strings are passed, instead of the original types. Is there any reason, why you insist on using strings?

**Comment by dwarakaprasad** [comment]
Thank you for your review comment @gzsombor.
I actually did start with using the original type but the Instant, LocalDate & ZonedDateTime were not getting decoded in the query and hence was causing syntax error (this was with postgress). All the field types were causing issues with H2 (syntax error) and hence had to settle with string for everything.

This code is tested with all the supported db's in the integration tests see jhipster/generator-jhipster#19959

**Comment by DanielFran** [comment]
@gzsombor Is it OK to merge it or should we investigate more the toString?

**Review by DanielFran — APPROVED** [review]

**Pull request merged**